DataDog/datadog-agent
# Host Metadata payload high priority

### What does this PR do?

This PR introduces the concept of transaction priority. 
A transaction with an high priority is dropped from the retry queue after all other transactions are dropped.
The host metadata payload is sent with an high priority. 

### Motivation

If the host is not known by the backend and the backend has an outage, the metrics received by the backends before the host metadata payload are not tags correctly.

### Additional Notes

Note: This PR does not guaranty that the host metada payload is the first payload sent after an outage.

### Describe your test plan

- Use the following `datadog.yaml` configuration
```
dd_url: http://127.0.0.1:1234 
forwarder_retry_queue_max_size: 5
```
- start the agent.
- wait for 2 minutes.
- start a locally proxy on 127.0.0.1 port 1234 and dump the http requests.
- The request `POST /intake/?api_key=API_KEY`containing metadata payload should be dumped by the proxy.
